File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {       
        unordered_map<int,int>mp;
        int maxi=0,c=0;
        for(auto i:nums){
            mp[i]++;
        }
        unordered_map<int,int>mp1;
        for(int i=0;i<nums.size();i++){
            if(mp.find(nums[i]-1)==mp.end()){
                c=1;
                int val=nums[i];
                while(mp1.find(nums[i])==mp1.end() && mp.find(val+1)!=mp.end()){
                    c++;
                    val++;
                    
                }
                maxi=max(maxi,c);
                mp1[nums[i]]++;
            }
        }
        return maxi;
    }
};

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
        if(nums.length==0){
            return 0;
        }
        HashMap<Integer,Integer>mp=new HashMap<>();
        for(int i:nums){
            mp.put(i,mp.getOrDefault(i,0)+1);
        }
        int maxi=Integer.MIN_VALUE;
        for(int i:mp.keySet()){
            int c=0;
            if(!mp.containsKey(i-1)){
                int j=i;
                while(mp.containsKey(j)){
                    j++;
                    c++;
                }
                maxi=Math.max(maxi,c);
            }

        }
        return maxi;

    }
}